src/lib/server/xlat_func.c

```diff
@@ -210,18 +210,41 @@ static ssize_t xlat_strlen(UNUSED TALLOC_CTX *ctx, char **out, size_t outlen,
 /** Print the size of the attribute in bytes.
  *
  */
-static ssize_t xlat_length(UNUSED TALLOC_CTX *ctx, char **out, size_t outlen,
-			   UNUSED void const *mod_inst, UNUSED void const *xlat_inst,
-			   REQUEST *request, char const *fmt)
+static xlat_action_t xlat_length(UNUSED TALLOC_CTX *ctx, fr_cursor_t *out,
+				REQUEST *request, UNUSED void const *xlat_inst,
+				UNUSED void *xlat_thread_inst, fr_value_box_t **in)
+
 {
-	VALUE_PAIR *vp;
+	VALUE_PAIR 	*vp;
+	fr_value_box_t	*vb;
 
-	while (isspace((int) *fmt)) fmt++;
+	/*
+	 * Concatenate all input
+	 */
+	if (fr_value_box_list_concat(ctx, *in, in, FR_TYPE_STRING, true) < 0) {
+		RPEDEBUG("Failed concatenating input");
+		return XLAT_ACTION_FAIL;
+	}
 
-	if ((xlat_fmt_get_vp(&vp, request, fmt) < 0) || !vp) return 0;
+	MEM(vb = fr_value_box_alloc(ctx, FR_TYPE_SIZE, NULL, false));
 
-	snprintf(*out, outlen, "%zu", fr_value_box_network_length(&vp->data));
-	return strlen(*out);
+	vb->vb_size = 0;
+
+	if (*in) {
+		char const *var = (*in)->vb_strvalue;
+
+		if ((xlat_fmt_get_vp(&vp, request, var) < 0) || !vp) {
+			talloc_free(vb);
+			return XLAT_ACTION_FAIL;
+		}
+
+		vb->vb_size = fr_value_box_network_length(&vp->data);
+	}
+
+	fr_cursor_insert(out, vb);
+	fr_cursor_append(out, vb);
```

Comment:
not sure we need both "insert" and "append" here.

Reply:
@alandekok It was a typo. I pushed the `fixup`

Reply:
@alandekok if everything is ok, let me know to rebase/update the pr.